NFD master panics when no matchFeatures are defined in NodeFeatureRule

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

```sh
I0421 13:44:17.450121       1 nfd-master.go:594] will process all nodes in the cluster
panic: assignment to entry in nil map

goroutine 163 [running]:
sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1.(*Features).InsertAttributeFeatures(...)
        /go/node-feature-discovery/pkg/apis/nfd/v1alpha1/feature.go:62
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).processNodeFeatureRule(0xc00080e8c0, 0xc000040720)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:924 +0x9a9
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).refreshNodeFeatures(0xc00080e8c0, 0x0?, {0xc0007ae180, 0x2f}, 0xc000803360?, 0xc000d65790?, 0x857f38?)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:746 +0xdc
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateOneNode(0xc00080e8c0, {0xc0007ae180, 0x2f})
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:675 +0x487
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateAllNodes(0xc00080e8c0)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:607 +0x170
sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).nfdAPIUpdateHandler(0xc00080e8c0)
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:335 +0x1cc
created by sigs.k8s.io/node-feature-discovery/pkg/nfd-master.(*nfdMaster).Run
        /go/node-feature-discovery/pkg/nfd-master/nfd-master.go:225 +0x6fe
```

**What you expected to happen**:

Log error or apply extended resources 😄 

**How to reproduce it (as minimally and precisely as possible)**:

```yaml
apiVersion: nfd.k8s-sigs.io/v1alpha1
kind: NodeFeatureRule
metadata:
  name: rdt-l3-closid-extended-rule
spec:
  rules:
    - name: "rdt l3 closid extended rule"
      extendedResources:
        feature.node.kubernetes.io/rdtl3_closid: "@cpu.rdt.RDTL3CA_NUM_CLOSID"
```

I have also tried it with empty list, still panicking:
```yaml
apiVersion: nfd.k8s-sigs.io/v1alpha1
kind: NodeFeatureRule
metadata:
  name: rdt-l3-closid-extended-rule
spec:
  rules:
    - name: "rdt l3 closid extended rule"
      extendedResources:
        feature.node.kubernetes.io/rdtl3_closid: "@cpu.rdt.RDTL3CA_NUM_CLOSID"
      matchFeatures: []
```

**Anything else we need to know?**:

Running from `v0.13.0` tag

**Environment**:
- Kubernetes version (use `kubectl version`): 1.23

--- a/pkg/apis/nfd/v1alpha1/feature.go
+++ b/pkg/apis/nfd/v1alpha1/feature.go
@@ -16,6 +16,15 @@ limitations under the License.
 
 package v1alpha1
 
+// NewNodeFeatureSpec creates a new emprty instance of NodeFeatureSpec type,
+// initializing all fields to proper empty values.
+func NewNodeFeatureSpec() *NodeFeatureSpec {
+	return &NodeFeatureSpec{
+		Features: *NewFeatures(),
+		Labels:   make(map[string]string),
+	}
+}
+
 // NewFeatures creates a new instance of Features, initializing all feature
 // types (flags, attributes and instances) to empty values.
 func NewFeatures() *Features {
@@ -56,7 +65,11 @@ func NewInstanceFeature(attrs map[string]string) *InstanceFeature {
 }
 
 // InsertAttributeFeatures inserts new values into a specific feature.
+// TODO: add unit tests
 func (f *Features) InsertAttributeFeatures(domain, feature string, values map[string]string) {
+	if f.Attributes == nil {
+		f.Attributes = make(map[string]AttributeFeatureSet)
+	}
 	key := domain + "." + feature
 	if _, ok := f.Attributes[key]; !ok {
 		f.Attributes[key] = NewAttributeFeatures(values)
@@ -70,6 +83,7 @@ func (f *Features) InsertAttributeFeatures(domain, feature string, values map[st
 
 // Exists returns a non-empty string if a feature exists. The return value is
 // the type of the feautre, i.e. "flag", "attribute" or "instance".
+// TODO: add unit tests
 func (f *Features) Exists(name string) string {
 	if _, ok := f.Flags[name]; ok {
 		return "flag"
@@ -85,6 +99,7 @@ func (f *Features) Exists(name string) string {
 
 // MergeInto merges two FeatureSpecs into one. Data in the input object takes
 // precedence (overwrite) over data of the existing object we're merging into.
+// TODO: add unit tests
 func (in *NodeFeatureSpec) MergeInto(out *NodeFeatureSpec) {
 	in.Features.MergeInto(&out.Features)
 	if in.Labels != nil {
@@ -100,6 +115,7 @@ func (in *NodeFeatureSpec) MergeInto(out *NodeFeatureSpec) {
 // MergeInto merges two sets of features into one. Features from the input set
 // take precedence (overwrite) features from the existing features of the set
 // we're merging into.
+// TODO: add unit tests
 func (in *Features) MergeInto(out *Features) {
 	if in.Flags != nil {
 		if out.Flags == nil {
@@ -134,6 +150,7 @@ func (in *Features) MergeInto(out *Features) {
 }
 
 // MergeInto merges two sets of flag featues.
+// TODO: add unit tests
 func (in *FlagFeatureSet) MergeInto(out *FlagFeatureSet) {
 	if in.Elements != nil {
 		if out.Elements == nil {
@@ -146,6 +163,7 @@ func (in *FlagFeatureSet) MergeInto(out *FlagFeatureSet) {
 }
 
 // MergeInto merges two sets of attribute featues.
+// TODO: add unit tests
 func (in *AttributeFeatureSet) MergeInto(out *AttributeFeatureSet) {
 	if in.Elements != nil {
 		if out.Elements == nil {
@@ -158,6 +176,7 @@ func (in *AttributeFeatureSet) MergeInto(out *AttributeFeatureSet) {
 }
 
 // MergeInto merges two sets of instance featues.
+// TODO: add unit tests
 func (in *InstanceFeatureSet) MergeInto(out *InstanceFeatureSet) {
 	if in.Elements != nil {
 		if out.Elements == nil {

--- a/pkg/nfd-master/nfd-master.go
+++ b/pkg/nfd-master/nfd-master.go
@@ -649,7 +649,8 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 
 	klog.V(1).Infof("processing node %q, initiated by NodeFeature API", nodeName)
 
-	features := &nfdv1alpha1.NodeFeatureSpec{}
+	features := nfdv1alpha1.NewNodeFeatureSpec()
+
 	annotations := Annotations{}
 
 	if len(objs) > 0 {

